### PR TITLE
Release Red03

### PR DESCRIPTION
**Pull request by SeveralCircles**
This release adds a variety of dumb new features and improves on existing ones. For changes see Files Changed.

Biggest change is addition of Colors Protection, for details on that see #16.

### CODE REVIEW

**Comment by SeveralCircles** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity increasing per file
==============================
- src/com/severalcircles/colors/commands/CommandSans.java  2
- src/com/severalcircles/colors/events/ChatEvent.java  2
- src/com/severalcircles/colors/system/protection/BuildExpiry.java  6
- src/com/severalcircles/colors/system/protection/ProtectionCheckFailException.java  2
- src/com/severalcircles/colors/commands/CommandNedry.java  4
- src/com/severalcircles/colors/system/Main.java  6
- src/com/severalcircles/colors/system/protection/RunViaSpigot.java  1
- src/com/severalcircles/colors/commands/CommandPersona.java  1
- src/com/severalcircles/colors/system/protection/ProtectionCheck.java  3
         

Clones added
============
- src/com/severalcircles/colors/system/protection/BuildExpiry.java  1
- src/com/severalcircles/colors/system/Main.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/aidanveney/colors/pullRequest?prid=5544530&bid=18177566)

**Comment by SeveralCircles** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity increasing per file
==============================
- src/com/severalcircles/colors/commands/CommandSans.java  2
- src/com/severalcircles/colors/events/ChatEvent.java  2
- src/com/severalcircles/colors/commands/CommandPersona.java  1
- src/com/severalcircles/colors/system/protection/ProtectionCheck.java  3
- src/com/severalcircles/colors/system/protection/BuildExpiry.java  6
- src/com/severalcircles/colors/system/Main.java  6
- src/com/severalcircles/colors/commands/CommandNedry.java  4
- src/com/severalcircles/colors/system/protection/ProtectionCheckFailException.java  2
- src/com/severalcircles/colors/system/protection/RunViaSpigot.java  1
         

Clones added
============
- src/com/severalcircles/colors/system/protection/BuildExpiry.java  1
- src/com/severalcircles/colors/system/Main.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/aidanveney/colors/pullRequest?prid=5544530&bid=18177566)

File: src/com/severalcircles/colors/commands/CommandNedry.java [hunk 1]
```diff
@@ -1,18 +1,37 @@
 package com.severalcircles.colors.commands;
```
Comment (SeveralCircles) [inline src/com/severalcircles/colors/commands/CommandNedry.java:1]: This file could really use some cleaning up. It's a bit of a mess.

File: src/com/severalcircles/colors/commands/CommandPersona.java [hunk 2]
```diff
@@ -5,13 +5,19 @@
...
 public class CommandPersona implements CommandExecutor {
     @Override
     public boolean onCommand(CommandSender commandSender, Command command, String s, String[] args) {
+        Player player = (Player) commandSender;
         int chatspam = 0;
-        while (chatspam<3000) {
-            Bukkit.broadcastMessage(ChatColor.DARK_RED + "Joker Persona");
-            chatspam++;
+        if (player.hasPermission("com.severalcircles.colors.persona")) {
+            while (chatspam < 3000) {
```
Comment (SeveralCircles) [inline src/com/severalcircles/colors/commands/CommandPersona.java]: We could reduce this number a bit, the messages fly by fast enough that it just doesn't matter. @Minotaurinmaze

**Review by Minotaurinmaze — CHANGES_REQUESTED** [review]
The "access" command is not properly written on the plugin.yml file. It should include <player> to let people using the command know that they can target specific players.

File: src/com/severalcircles/colors/commands/CommandSans.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Player;
+
+public class CommandSans implements CommandExecutor {
+    @Override
+    public boolean onCommand(CommandSender commandSender, Command command, String s, String[] args) {
+        Player player = (Player) commandSender;
+        Player target = Bukkit.getPlayer(args[0]);
+        if (player.hasPermission("com.severalcircles.colors.sans")) {
+            target.sendMessage(ChatColor.BLUE + "getttttttttttt dunked on!");
```
Comment (SeveralCircles) [inline src/com/severalcircles/colors/commands/CommandSans.java:17]: This message doesn't make a lot of sense for sans. Requesting more sans. @Minotaurinmaze

**Comment by SeveralCircles** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity increasing per file
==============================
- src/com/severalcircles/colors/system/protection/BuildExpiry.java  6
- src/com/severalcircles/colors/commands/CommandNedry.java  4
- src/com/severalcircles/colors/system/protection/RunViaSpigot.java  1
- src/com/severalcircles/colors/commands/CommandSans.java  2
- src/com/severalcircles/colors/system/protection/ProtectionCheckFailException.java  2
- src/com/severalcircles/colors/commands/CommandPersona.java  1
- src/com/severalcircles/colors/system/Main.java  6
- src/com/severalcircles/colors/events/ChatEvent.java  2
- src/com/severalcircles/colors/system/protection/ProtectionCheck.java  3
         

Clones added
============
- src/com/severalcircles/colors/system/protection/BuildExpiry.java  1
- src/com/severalcircles/colors/system/Main.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/aidanveney/colors/pullRequest?prid=5544530&bid=18177566)

**Review by BeegBrainMo — CHANGES_REQUESTED** [review]
@Minotaurinmaze Please look at what @SeveralCircles asked you to fix

**Review by Minotaurinmaze — APPROVED** [review]
Everything looks good! I don't see any current issues with it.

**Comment by SeveralCircles** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- src/com/severalcircles/colors/commands/CommandPersona.java  1
- src/com/severalcircles/colors/commands/CommandSans.java  2
- src/com/severalcircles/colors/system/protection/ProtectionCheckFailException.java  1
- src/com/severalcircles/colors/events/ChatEvent.java  2
- src/com/severalcircles/colors/system/protection/ProtectionCheck.java  3
- src/com/severalcircles/colors/system/protection/BuildExpiry.java  6
- src/com/severalcircles/colors/commands/CommandNedry.java  3
- src/com/severalcircles/colors/system/protection/RunViaSpigot.java  1
- src/com/severalcircles/colors/system/Main.java  6
         

Clones added
============
- src/com/severalcircles/colors/system/protection/BuildExpiry.java  1
- src/com/severalcircles/colors/system/Main.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/aidanveney/colors/pullRequest?prid=5544530&bid=18177566)

**Review by Minotaurinmaze — APPROVED** [review]
I did an uh oh and re-requested review. The code looks good. I don't see anything wrong with it.

**Review by BeegBrainMo — APPROVED** [review]
👌Yes, the changes are very spicy.

**Review by SeveralCircles — COMMENTED** [review]
All looks good to me!

**Comment by SeveralCircles** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/com/severalcircles/colors/commands/CommandNedry.java  3
- src/com/severalcircles/colors/events/ChatEvent.java  2
- src/com/severalcircles/colors/commands/CommandSans.java  2
- src/com/severalcircles/colors/system/protection/ProtectionCheck.java  3
- src/com/severalcircles/colors/system/protection/RunViaSpigot.java  1
- src/com/severalcircles/colors/system/Main.java  6
- src/com/severalcircles/colors/system/protection/ProtectionCheckFailException.java  1
- src/com/severalcircles/colors/system/protection/BuildExpiry.java  6
- src/com/severalcircles/colors/commands/CommandPersona.java  1
         

Clones added
============
- src/com/severalcircles/colors/system/Main.java  1
- src/com/severalcircles/colors/system/protection/BuildExpiry.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/aidanveney/colors/pullRequest?prid=5544530&bid=18177566)

**Pull request merged**